pkg/porter/credentials.go

```diff
@@ -0,0 +1,90 @@
+package porter
+
+import (
+	"fmt"
+
+	"github.com/deislabs/porter/pkg/credentials"
+	"github.com/deislabs/porter/pkg/printer"
+	"github.com/pkg/errors"
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+func (p *Porter) PrintCredentials(opts printer.PrintOptions) error {
+	return nil
+}
+
+type CredentialOptions struct {
+	sharedOptions
+	DryRun bool
+	Silent bool
+}
+
+// Validate prepares for an action and validates the options.
+// For example, relative paths are converted to full paths and then checked that
+// they exist and are accessible.
+func (g *CredentialOptions) Validate(args []string) error {
+	err := g.validateCredName(args)
+	if err != nil {
+		return err
+	}
+
+	err = g.validateBundlePath()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (g *CredentialOptions) validateCredName(args []string) error {
+	if len(args) == 1 {
+		g.Name = args[0]
+	} else if len(args) > 1 {
+		return errors.Errorf("only one positional argument may be specified, the credential name, but multiple were received: %s", args)
+	}
+	return nil
+}
+
+func (p *Porter) GenerateCredentials(opts CredentialOptions) error {
+
+	//TODO make this work for either porter.yaml OR a bundle
```

Comment:
+1

Reply:
Yup, per our goals from #320, all of our commands should stop taking bundle.json files as arguments when a porter bundle is present and work off of the manifest first. But for now this is a great first step! 💯